AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Setreg.py

```diff
@@ -0,0 +1,109 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+"""
+
+import pytest
+import os
+import sys
+from ly_test_tools.o3de.editor_test import EditorTestSuite
+
+from .utils.FileManagement import FileManagement as fm
+
+sys.path.append(os.path.dirname(os.path.abspath(__file__)) + '/../automatedtesting_shared')
+
+import ly_test_tools.environment.process_utils as       process_utils
+import ly_test_tools.launchers.launcher_helper as       launcher_helper
+from   ly_test_tools.log.log_monitor           import   LogMonitor
+import ly_test_tools.environment.waiter        as       waiter
+
+@pytest.mark.SUITE_sandbox
+@pytest.mark.parametrize("project", ["AutomatedTesting"])
+@pytest.mark.parametrize("launcher_platform", ['windows_editor'])
+class TestAutomation(EditorTestSuite):
+
+	def _setup(self, workspace, target_file, src_file, parent_path=None, search_subdirs=False):
+		"""
+		Inner function to wrap around decorated function. Function being decorated must match this
+		functions parameter order.
+		"""
+		self.target_file = target_file
+
+		root_path = parent_path
+		if root_path is not None:
+			root_path = os.path.join(workspace.paths.engine_root(), root_path)
+		else:
+			# Default to project folder (AutomatedTesting)
+			root_path = workspace.paths.project()
+
+		# Try to locate both target and source files
+		try:
+			self.file_list = fm._find_files([target_file, src_file], root_path, search_subdirs)
+		except RuntimeWarning as w:
+			assert False, (
+				w.args[0]
+				+ " Please check use of search_subdirs; make sure you are using the correct parent directory."
+			)
+
+		fm._restore_file(target_file, self.file_list[target_file])
+		fm._backup_file(target_file, self.file_list[target_file])
+		fm._copy_file(src_file, self.file_list[src_file], target_file, self.file_list[target_file])
+		# Run the test now.
+		
+	def _teardown(self):
+		"""
+		Cleanup
+		"""
+		fm._restore_file(self.target_file, self.file_list[self.target_file])
+
+	def test_Multiplayer_SimpleGameServerLauncher_ConnectsSuccessfully_WithSetRegConfig(self, workspace, launcher_platform, crash_log_watchdog):
+		"""
+		"""
+		try:
+			self._setup(workspace, 'autoexec.server.setreg','autoexec.server_host.setreg_override','AutomatedTesting/Registry', search_subdirs=True)
```

Comment:
It would safer to copy the test files in the AutomatedTesting/user/Registry folder since that is a transient folder.

Also I would suggest having naming the copied setreg files in a way that a normal user wouldn't be able to accidentally create.
For example if the user has a modified `autoexec.server.setreg` in their AutomatedTesting/Registry directory, this would override it.
How about `multiplayer_test.server.setreg`?

Reply:
I am following the pattern of these files as used by physx tests, for example: `C:\git\o3de\AutomatedTesting\Registry\physx_overrides\Material_CharacterController.setreg_override`

Reply:
That pattern they are using is not really a great pattern to use.

Alternatively what can be done is to use the` --regset-file` option and avoid having to copy the file files around.